### DISCUSSION TITLE
Standard Input (stdin)

### DISCUSSION THREAD

**Discussion by scholes231** [sshnet/SSH.NET#1340] [General]
I'd like to send file via ssh. My command: addFile -s filename < C:\...\....txt. Is this supported?
I've tried this but not worked:
using (var shell = client.CreateShellStream("dumb", 0, 0, 0, 0, 4096))
{
    var commandWithInput = $"{command}\n";
    using (var inputMemoryStream = new MemoryStream(Encoding.UTF8.GetBytes(commandWithInput)))
    {
        inputMemoryStream.CopyTo(shell);
    }

    using (var fileStream = new MemoryStream(input))
    {
        fileStream.CopyTo(shell);
    }

    result = new StreamReader(shell).ReadToEnd();
    _logger.LogInformation($"SSH Command result: {result}");
}
Where:
command variable: "addFile -s filename"
input variable  is byte array

I have exception handling, but no error, result variable is empty. What did i wrong?

**Comment by Rob-Hague** [comment]
I have 2 suggestions:

1. Try `shell.Flush()` after the `CopyTo` calls
2. Try instead using `SshCommand` as in here:
https://github.com/sshnet/SSH.NET/blob/c044fd40c647b39768cc412b6a37bdc151d28d9e/test/Renci.SshNet.IntegrationTests/SshClientTests.cs#L25-L45

**Reply by scholes231** [reply]
Thank you for your quick reply. I'll try

**Reply by scholes231** [reply]
I've tried, but not working. After EndExecute is nothing happened. Command is not completed. What's the problem?
